libwallet/assets/dcr/consensus.go

```diff
@@ -111,22 +110,19 @@ func (asset *Asset) SetVoteChoice(agendaID, choiceID, hash, passphrase string) e
 		return err
 	}
 
-	currentChoice := w.AgendaChoice{
-		AgendaID: agendaID,
-		ChoiceID: "abstain", // default to abstain as current choice if not found in wallet
+	choice, ok := choices[agendaID]
+	if ok && choice == strings.ToLower(choiceID) {
+		// Do not set the same choice again
+		return nil
 	}
 
-	for i := range choices {
-		if choices[i].AgendaID == agendaID {
-			currentChoice.ChoiceID = choices[i].ChoiceID
-			break
-		}
+	if !ok {
+		// Default to abstain if no previous choice existed
+		choice = "abstain"
 	}
 
-	newChoice := w.AgendaChoice{
-		AgendaID: agendaID,
-		ChoiceID: strings.ToLower(choiceID),
-	}
+	currentChoice := map[string]string{agendaID: choice}
+	newChoice := map[string]string{agendaID: strings.ToLower(choiceID)}
```

Comment:
currentChoices and newChoices?

A map can't really represent one object only, even if it has only one item in it.